lib/hanami/providers/db.rb

```diff
@@ -27,11 +33,23 @@ def prepare
           raise Hanami::ComponentLoadError, "A database_url is required to start :db."
         end
 
-        @rom_config = ROM::Configuration.new(config.adapter, database_url, extensions: config.extensions)
+        connection = self.class.connection_cache[database_url]
```

Comment:
Because this connection cache needs to cut across _all_ containers (i.e. slices) in the Hanami app. Registering it in one particular container doesn't really help here, we need it to live in exactly one place globally. Plus, it's an internal implementation detail, not a component that we will expect a user to use.

Reply:
Plus, because this cache will hold the connection objects for _all slices in the app_, it should never be registered into any single slice, because that would break the encapsulation of the slices themselves - by virtue of having that object registered, the slice would then suddenly know far too much about the outside world.